wandb/sdk/wandb_run.py

```diff
@@ -2890,6 +2890,13 @@ def link_artifact(
                     wandb.termwarn(
                         "Artifact TTL will be disabled for source artifacts that are linked to portfolios."
                     )
+                result = handle.wait(timeout=-1)
+                if result is None:
+                    handle.abandon()
+                assert result is not None
+                response = result.response.link_artifact_response
+                if response.error_message:
+                    wandb.termerror(response.error_message)
```

Comment:
suggestion: consider avoiding `assert` here too if possible -- would it be possible to put the latter block inside an `else` clause?
```suggestion
                result = handle.wait(timeout=-1)
                if result is None:
                    handle.abandon()
                else:
                    response = result.response.link_artifact_response
                    if response.error_message:
                        wandb.termerror(response.error_message)
```